internal/errorcodes/errorcodes.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieout.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errorcodes
+
+import (
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+type Reason string
+
+const (
+	ErrorCodesFeatureMetadataKey = "error_codes_feature"
+	Owner                        = "components-contrib"
+	Domain                       = "dapr.io"
+)
+
+var StateETagMismatchReason = Reason("DAPR_STATE_ETAG_MISMATCH")
+
+type ResourceInfoData struct {
+	ResourceType, ResourceName string
+}
+
+func FeatureEnabled(md map[string]string) bool {
+	if _, ok := md[ErrorCodesFeatureMetadataKey]; ok {
```

Comment:
As per comment on other PR, our metadata parsing should ideally be case-insensitive.

Reply:
As I mentioned in the other PR, will be shared one both PRs are merged.